client/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+import { OutlinedInput, InputLabel, InputAdornment } from "@material-ui/core";
+import LocationOnIcon from "@material-ui/icons/LocationOn";
+import LocationOffIcon from "@material-ui/icons/LocationOff";
+import { isDebug } from "src/config";
+
+function SearchBar({
+  iconCallback,
+  onChangeCallback,
+  onEnterCallback,
+}: {
+  iconCallback?: (status: boolean) => void;
+  onChangeCallback?: (text: string) => void;
+  onEnterCallback?: (text: string) => void;
+}) {
+  const [gpsStatus, setGpsStatus] = useState(false);
+
+  const iconClickWrapper = () => {
+    setGpsStatus(!gpsStatus);
+  };
+
+  const dynamicChangeWrapper = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    if (onChangeCallback) {
+      onChangeCallback(event.target.value);
+    }
+  };
+
+  const keyPressWrapper = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === "Enter") {
+      const ele = document.getElementById("stores-search-bar");
+      if (ele) ele.blur();
+    }
+  };
+
+  const blurWrapper = (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    if (onEnterCallback) {
+      onEnterCallback(event.target.value);
```

Comment:
should this be the keyPressWrapper?

Reply:
My intended functionality is that after user types, then press ENTER, two things should happen:
1) Unfocus current Input field
2) Fire off the callback to submit query
The difference between onChange and onEnter is that the onChange would dynamically filter present search results
And onEnter will fire off a server call to retrieve new results
I've currently used the blur event to do (2) since (1) will fire this and on mobile, users may not press ENTER but tap elsewhere to dismiss the Type dialog
Perhaps the confusion here is the naming of `onEnterCallback`. I have changed this to `onSubmitCallback` to reflect that user submits 'confirmed' search query

Reply:
got it! renaming sounds good, clarifies the intention